autoemulate/experimental/emulators/transformed/base.py

```diff
@@ -0,0 +1,230 @@
+from typing import cast
+
+import torch
+from linear_operator.operators import DiagLinearOperator
+from torch.distributions import (
+    ComposeTransform,
+    Transform,
+    TransformedDistribution,
+)
+
+from autoemulate.experimental.data.utils import ValidationMixin
+from autoemulate.experimental.device import TorchDeviceMixin
+from autoemulate.experimental.emulators.base import Emulator
+from autoemulate.experimental.transforms.base import AutoEmulateTransform
+from autoemulate.experimental.transforms.utils import make_positive_definite
+from autoemulate.experimental.types import (
+    DeviceLike,
+    DistributionLike,
+    GaussianLike,
+    OutputLike,
+    TensorLike,
+)
+
+
+class TransformedEmulator(Emulator, ValidationMixin):
+    transforms: list[AutoEmulateTransform]
+    model: Emulator
+    target_transforms: list[AutoEmulateTransform]
+
+    def __init__(  # noqa: PLR0913
+        self,
+        x: TensorLike,
+        y: TensorLike,
+        transforms: list[AutoEmulateTransform] | None,
+        target_transforms: list[AutoEmulateTransform] | None,
+        model: type[Emulator],
+        output_from_samples: bool = False,
+        n_samples: int = 1000,
+        full_covariance: bool = False,
+        max_targets: int = 200,
+        device: DeviceLike = "cpu",
+        **kwargs,
+    ):
+        """Initialize a transformed emulator.
+
+        Parameters
+        ----------
+            x (TensorLike): Input data tensor.
```

Comment:
This is super minor and I feel bad about even mentioning but related to #543, we should agree exactly on how to write docstrings and stick to it throughout. 

For example I do:
```
x: TensorLike
    Input data tensor.
```

I don't have a strong preference, so lets pick one and maybe open an issue to make this consistent across the code base (somewhere down the line when someone has some spare time :sweat_smile:).

Reply:
Thanks for mentioning this - I completely agree it would be great to be consistent and also include in the style guide! It looks like the current [CONTRIBUTING.md](https://github.com/alan-turing-institute/autoemulate/blob/b4faf29cf5671559163a6e4aa4d45295f7bdd314/CONTRIBUTING.md) indicates [numpy-style docstrings](https://numpydoc.readthedocs.io/en/latest/format.html) so I could update to these throughout this PR and I can open an issue to specifically revisit this?

Reply:
Sounds great!